### PR TITLE
models: Switch to kornia AugmentationSequential

### PR DESCRIPTION
**Pull request by ashnair1**
Switch to kornia's `AugmentationSequential` in `models` submodule

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you resolve the merge conflicts?

File: torchgeo/models/swin.py [hunk 1]
```diff
@@ -31,17 +29,17 @@
     [255.0, 255.0, 255.0, 8160.0, 8160.0, 8160.0, 8160.0, 8160.0, 8160.0]
 )  # noqa: E501
 _mean = torch.zeros_like(_std)
-_sentinel2_ms_satlas_transforms = AugmentationSequential(
+_sentinel2_ms_satlas_transforms = K.AugmentationSequential(
     K.Normalize(mean=_mean, std=_std),
-    Lambda(lambda x: torch.clamp(x, min=0.0, max=1.0)),
-    data_keys=["image"],
+    Lambda(lambda x: torch.clamp(x, min=0.0, max=1.0)),  # type: ignore[arg-type]
```
Comment (adamjstewart) [inline torchgeo/models/swin.py:34]: I think we decided we're supposed to wrap Lambda in an ImageSequential in order to get the style tests to pass?

**Review by adamjstewart — COMMENTED** [review]
There are likely many lines in `pyproject.toml` pytest ignores that could be removed now that we're bumping the min version. Same with all of the resame/align_corners stuff in the data modules. The former we could do now. The latter maybe in a separate PR that updates the data modules?

File: torchgeo/models/dofa.py [hunk 2]
```diff
@@ -375,7 +373,7 @@ def forward(self, x: Tensor, wavelengths: list[float]) -> Tensor:
 
 # https://github.com/zhu-xlab/DOFA/blob/master/normalize_dataset.py
 # Normalization is sensor-dependent and is therefore left out
-_dofa_transforms = AugmentationSequential(K.CenterCrop((224, 224)), data_keys=["image"])
+_dofa_transforms = K.AugmentationSequential(K.CenterCrop((224, 224)), data_keys=None)
```
[the remark this answers is not in the document]
Comment (johnnv1) [reply torchgeo/models/dofa.py:376]: "Activate"/allow the usage of dictionaries. The API will try wrap the dictionaries only if don't have data keys

**Review by johnnv1 — COMMENTED** [review]
Cool patch. What still has the necessity of the augmentation sequential torch geo wrap?

File: pyproject.toml [hunk 3]
```diff
@@ -259,8 +257,6 @@ filterwarnings = [
     "ignore:Starting from v1.9.0, `tensorboardX` has been removed as a dependency of the `lightning.pytorch` package:UserWarning",
     # https://github.com/Lightning-AI/lightning/issues/18545
     "ignore:LightningCLI's args parameter is intended to run from within Python like if it were from the command line.:UserWarning",
-    # https://github.com/kornia/kornia/pull/1611
-    "ignore:`ColorJitter` is now following Torchvision implementation.:DeprecationWarning:kornia.augmentation._2d.intensity.color_jitter",
     # https://github.com/kornia/kornia/pull/1663
```
Comment (adamjstewart) [inline pyproject.toml:260]: I'm guessing this warning is still active even in the latest release?

**Review by adamjstewart — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (ashnair1) [reply pyproject.toml:260]: Yes it is.

Comment (ashnair1) [reply pyproject.toml:260]: Yes, it is.

**Pull request merged**